e2e-tests/self-upgrade-no-symlink.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+unset CARGO_INSTALL_ROOT
+
+export CARGO_HOME=$(mktemp -d 2>/dev/null || mktemp -d -t 'cargo-home')
+export PATH="$CARGO_HOME/bin:$PATH"
+
+# first boostrap-install into the CARGO_HOME
+mkdir -p "$CARGO_HOME/bin"
+cp "./$1" "$CARGO_HOME/bin"
+
+# now we're running the CARGO_HOME/bin/cargo-binstall (via cargo):
+
+# self update replacing no-symlinks with no-symlinks
+cargo binstall --no-confirm --no-symlinks --force cargo-binstall
+
+# self update replacing no-symlinks with symlinks
+cp "./$1" "$CARGO_HOME/bin"
+
+cargo binstall --no-confirm --force cargo-binstall
+
+# self update replacing symlinks with symlinks
+ln -sf "./$1" "$CARGO_HOME/bin/cargo-binstall"
+
+cargo binstall --no-confirm --force cargo-binstall
```

Comment:
This failed on CI but runs just fine on my M1.

Reply:
They all failed with:

```
+ cargo binstall --no-confirm --force cargo-binstall
error: no such command: `binstall`

	Did you mean `install`?

	View all installed commands with `cargo --list`
```

seems like `ln -sf ./cargo-binstall "$CARGO_HOME/bin/cargo-binstall"` is the one to blame here, but IDK what happened.

Reply:
oh, yeah, that `ln` call is wrong. that created a link pointing to itself. use the full path:

```
ln -snf "$(pwd)/cargo-binstall" "$CARGO_HOME/bin/cargo-binstall"
```